app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -151,18 +166,20 @@ def apply_common_props(button, input)
       button[:url_parms] = "?show=#{request.parameters[:show]}"
     end
 
-    dis_title = build_toolbar_disable_button(button[:child_id] || button[:id])
+    dis_title = disable_button(button[:child_id] || button[:id])
     if dis_title
       button[:enabled] = false
       if dis_title.kind_of? String
         button[:title] = button.localized(:title, dis_title)
       end
     end
+    button.calculate_properties
```

Comment:
I would put this line inside an else branch of the above if statement, since ```#calculate_properties``` checks whether the button is disabled and sets the enabled status appropriately. Now, ```#calculate_properties``` is executed regardless of the ```dis_title```'s value and reassigns the enabled status of the button, hence making the line 171 unnecessary.